spring-integration-http/src/main/java/org/springframework/integration/http/outbound/HttpRequestExecutingMessageHandler.java

```diff
@@ -360,8 +360,10 @@ protected void doInit() {
 			ConverterRegistry converterRegistry =
 					(ConverterRegistry) conversionService;
 
-			converterRegistry.addConverter(new ClassToStringConverter());
-			converterRegistry.addConverter(new ObjectToStringConverter());
+			if (!conversionService.canConvert(Class.class, String.class)) {
```

Comment:
Oleg introduced this: https://jira.spring.io/browse/INT-1760
Gunnar changed that to this: https://jira.spring.io/browse/INT-2689
And I eventually added `Assert` for return object tyoe from SpEL.